pkg/resource/logging-credentials/reconciler.go

```diff
@@ -0,0 +1,100 @@
+package loggingcredentials
+
+import (
+	"context"
+	"fmt"
+
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler implements a reconciler.Interface to handle
+// Logging Credentials: store and maintain logging credentials
+type Reconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// ReconcileCreate ensures a secret exists for the given cluster.
+func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("loggingCredentials secret create")
+
+	const secretNamespace = "monitoring"
+	const secretName = "logging-credentials"
+
+	// Check if secrets exist.
+	logger.Info(fmt.Sprintf("loggingcredentials checking secret %s/%s", secretNamespace, secretName))
+
+	var currentSecret v1.Secret
+
+	// To be updated when secret is modified
+	var secretUpdated bool = false
+
+	err := r.Client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: secretNamespace}, &currentSecret)
+	if err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			logger.Info("logging-credentials not found, initializing one")
+			// Create basic secret
+			currentSecret, err = GenerateLoggingCredentialsBasicSecret(lc)
```

Comment:
Here we are getting and generating secret with different name/namespace.
Please make sure we use the same values, when reading and when generating the secret.

Reply:
I agree.
Is there such a thing as "package-global constants" that I could re-use across the package? Or should I write functions that will return the expected values?

Reply:
Nevermind, should be global because other reconcilers will need it too.
So I'll do it as a config we can pass at startup.

Reply:
Done.

Reply:
Changed it to retrieve info from `LoggingCredentialsSecretMeta(lc)`.
Yet I have some more tests to do because I currently have performance issues. :cry: